flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionFactory.java

```diff
@@ -157,8 +159,9 @@ private static void initializeBoundedBlockingPartitions(
 		int i = 0;
 		try {
 			for (; i < subpartitions.length; i++) {
-				subpartitions[i] = new BoundedBlockingSubpartition(
-					i, parent, ioManager.createChannel().getPathFile().toPath());
+				final File spillFile = ioManager.createChannel().getPathFile();
+				subpartitions[i] =
+						BoundedBlockingSubpartition.createWithFileAndMemoryMappedReader(i, parent, spillFile);
```

Comment:
Now that three ways of file writing/reading are implemented, why not make it configurable?

Reply:
I am not confident enough in any other implementation, yet.
I only added them for our testing and benchmarking.

I am also always unsure about adding too many configurable parts for users. For API / behavior, we need customizability to fit the application and setup needs.

For runtime and performance, I tend to favor one opinionated way that works well across the board. If there is another option that is 10% faster in 10% of the cases, I am not sure users benefit too much from that, but the added configuration makes it overall harder to understand and predict.